client/cache_test.go

```diff
@@ -0,0 +1,145 @@
+		object2 := unstructured.Unstructured{}
+		object2.SetName("cached-object2")
+		cache.CacheList(configMapGVK, "default", nil, []unstructured.Unstructured{object1, object2})
+
+		cachedObjects, err := cache.List(configMapGVK, "default", nil)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(cachedObjects).To(HaveLen(2))
+		Expect(cachedObjects[0]).To(Equal(object1))
+		Expect(cachedObjects[1]).To(Equal(object2))
+
+		// Simulate an object being deleted
+		cache.CacheList(configMapGVK, "default", nil, []unstructured.Unstructured{object2})
+
+		cachedObjects, err = cache.List(configMapGVK, "default", nil)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(cachedObjects).To(HaveLen(1))
+		Expect(cachedObjects[0]).To(Equal(object2))
+
+		cache.UncacheList(configMapGVK, "default", nil)
+		cachedObjects, err = cache.List(configMapGVK, "default", nil)
+		Expect(err).To(MatchError(ErrNoCacheEntry))
+		Expect(cachedObjects).To(BeNil())
+	})
+
+	It("It can convert a GVK to a GVR", func() {
+		gvr, err := cache.GVKToGVR(configMapGVK)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(gvr.Resource).To(Equal("configmaps"))
+	})
+
+	It("It fails when a resource is invalid when converting a GVK to a GVR", func() {
+		// Invalid group
+		_, err := cache.GVKToGVR(schema.GroupVersionKind{Group: "Italian Food", Version: "v1", Kind: "Spaghetti"})
+		Expect(err).To(MatchError(ErrNoVersionedResource))
+
+		// Valid group but invalid kind
+		_, err = cache.GVKToGVR(schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Spaghetti"})
+		Expect(err).To(MatchError(ErrNoVersionedResource))
+	})
+
+	It("It can have its cache cleared", func() {
+		gvr, err := cache.GVKToGVR(configMapGVK)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(gvr.Resource).To(Equal("configmaps"))
+
+		object1 := unstructured.Unstructured{}
+		object1.SetName("cached-object1")
+		object2 := unstructured.Unstructured{}
+		object2.SetName("cached-object2")
+		cache.CacheObject(configMapGVK, "default", "something1", &object1)
+		cache.CacheObject(configMapGVK, "default", "something2", &object2)
+
+		cache.Clear()
+
+		cachedObject1, err := cache.Get(configMapGVK, "default", "something1")
+		Expect(err).To(MatchError(ErrNoCacheEntry))
+		Expect(cachedObject1).To(BeNil())
+
+		cachedObject2, err := cache.Get(configMapGVK, "default", "something2")
+		Expect(err).To(MatchError(ErrNoCacheEntry))
+		Expect(cachedObject2).To(BeNil())
+	})
+
+	It("It can have an empty cache entry", func() {
+		cache.CacheList(configMapGVK, "default", nil, nil)
+
+		cachedObjects, err := cache.List(configMapGVK, "default", nil)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(cachedObjects).To(BeEmpty())
+
+		obj := ObjectIdentifier{
+			Group:     "",
+			Version:   "v1",
+			Kind:      "ConfigMap",
+			Namespace: "test-ns",
+			Name:      "watcher",
+		}
+		cache.CacheFromObjectIdentifier(obj, nil)
+
+		cachedObject, err := cache.Get(configMapGVK, "test-ns", "watcher")
+		Expect(err).ToNot(HaveOccurred())
```

Comment:
maybe return err? `// Get returns the object from the cache. The error ErrNoCacheEntry is returned if there is no cache entry.`

Reply:
Same explanation as above.